mactop: 0.2.3 -> 2.0.5, migrate to metaspartan/mactop

metaspartan opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- Tested, as applicable:
  - [X] [NixOS tests] in [nixos/tests].
  - [X] [Package tests] at `passthru.tests`.
  - [X] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [X] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

metaspartan committed:
mactop: 0.2.3 -> 2.0.5, migrate to metaspartan/mactop

mactop: 0.2.3 -> 2.0.5, migrate to metaspartan/mactop

dramforever reviewed (dismissed):
See [`pkgs/README.md`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md)

> Source-available software should be built from source where possible.

I find it unlikely that your program specifically depends on 1.25.4 and doesn't work with 1.25.5 (https://github.com/golang/go/compare/go1.25.4...go1.25.5), but even if it is the case, requiring 1.25.5 is not sufficient grounds to using a pre-built version of the program in Nixpkgs.

metaspartan commented:
> See [`pkgs/README.md`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md)
> 
> > Source-available software should be built from source where possible.
> 
> I find it unlikely that your program specifically depends on 1.25.4 and doesn't work with 1.25.5 ([golang/go@go1.25.4...go1.25.5](https://github.com/golang/go/compare/go1.25.4...go1.25.5)), but even if it is the case, requiring 1.25.5 is not sufficient grounds to using a pre-built version of the program in Nixpkgs.

Updated to build from source with a postpatch of 1.25.5 to 1.25.4 for now until fixed on repo. Thanks.

metaspartan commented:
> This seems fine to me, but I'll leave it to the package maintainer, who seems to be active, to decide

Any update? I updated this to fully build from source now, no more go v1.25 requirement and squashed the commits. Thanks
